activestorage/lib/active_storage/attached/changes/create_one.rb

```diff
@@ -70,6 +70,28 @@ def find_or_build_blob
           )
         when String
           ActiveStorage::Blob.find_signed!(attachable, record: record)
+        when File
+          filename = File.basename(attachable.path)
+          content_type = Marcel::MimeType.for(attachable, name: filename)
+
+          ActiveStorage::Blob.create_and_upload!(
+            io: attachable,
+            filename: filename,
+            content_type: content_type,
+            record: record,
+            service_name: attachment_service_name
+          )
+        when Pathname
+          filename = File.basename(attachable.basename)
+          content_type = Marcel::MimeType.for(attachable.open, name: filename)
```

Comment:
Does invoking `attachable.open` twice (once here, and once in the `io:` keyword argument below) pose any issues here?

Reply:
We should probably not call `open` twice as it'll use two file descriptors.

Reply:
I've removed the second call to open